spopt/tests/test_locate.py

```diff
@@ -0,0 +1,329 @@
+        self.assertEqual(mclp.problem.status, pulp.LpStatusInfeasible)
+
+    def test_optimality_pcenter_from_cost_matrix(self):
+        pcenter = PCenter.from_cost_matrix(
+            self.cost_matrix, self.ai, p_facilities=self.p_facility
+        )
+        pcenter = pcenter.solve(pulp.PULP_CBC_CMD())
+        self.assertEqual(pcenter.problem.status, pulp.LpStatusOptimal)
+
+    def test_infeasibility_pcenter_from_cost_matrix(self):
+        pcenter = PCenter.from_cost_matrix(self.cost_matrix, self.ai, p_facilities=0)
+        pcenter = pcenter.solve(pulp.PULP_CBC_CMD())
+        self.assertEqual(pcenter.problem.status, pulp.LpStatusInfeasible)
+
+    def test_optimality_pcenter_from_geodataframe(self):
+        pcenter = PCenter.from_geodataframe(
+            self.demand_points_gdf,
+            self.facility_points_gdf,
+            "geometry",
+            "geometry",
+            "POP2000",
+            p_facilities=self.p_facility,
+        )
+        pcenter = pcenter.solve(pulp.PULP_CBC_CMD())
+        self.assertEqual(pcenter.problem.status, pulp.LpStatusOptimal)
+
+    def test_infeasibility_pcenter_from_geodataframe(self):
+        pcenter = PCenter.from_geodataframe(
+            self.demand_points_gdf,
+            self.facility_points_gdf,
+            "geometry",
+            "geometry",
+            "POP2000",
+            p_facilities=0,
+        )
+        pcenter = pcenter.solve(pulp.PULP_CBC_CMD())
+        self.assertEqual(pcenter.problem.status, pulp.LpStatusInfeasible)
+
+    def test_optimality_pmedian_from_cost_matrix(self):
+        pmedian = PMedian.from_cost_matrix(
+            self.cost_matrix, self.ai, p_facilities=self.p_facility
+        )
+        pmedian = pmedian.solve(pulp.PULP_CBC_CMD())
+        self.assertEqual(pmedian.problem.status, pulp.LpStatusOptimal)
+
+    def test_infeasibility_pmedian_from_cost_matrix(self):
+        pmedian = PMedian.from_cost_matrix(self.cost_matrix, self.ai, p_facilities=0)
+        pmedian = pmedian.solve(pulp.PULP_CBC_CMD())
+        self.assertEqual(pmedian.problem.status, pulp.LpStatusInfeasible)
+
+    def test_optimality_pmedian_from_geodataframe(self):
+        pmedian = PMedian.from_geodataframe(
+            self.demand_points_gdf,
+            self.facility_points_gdf,
+            "geometry",
+            "geometry",
+            "POP2000",
+            p_facilities=self.p_facility,
+        )
+        pmedian = pmedian.solve(pulp.PULP_CBC_CMD())
+        self.assertEqual(pmedian.problem.status, pulp.LpStatusOptimal)
+
+    def test_infeasibility_pmedian_from_geodataframe(self):
+        pmedian = PMedian.from_geodataframe(
+            self.demand_points_gdf,
+            self.facility_points_gdf,
+            "geometry",
+            "geometry",
+            "POP2000",
+            p_facilities=0,
+        )
+        pmedian = pmedian.solve(pulp.PULP_CBC_CMD())
+        self.assertEqual(pmedian.problem.status, pulp.LpStatusInfeasible)
```

Comment:
Great job getting testing for `Global` and `Optimal` schema. Now let's add third and fourth schema: `Objective` and `Errors`. In `Objective` we will want to test for known objective values, optimal selection sets of problems, etc.  And in `Errors` we will want to test for raised exceptions and warnings, etc.